src/base.jl

```diff
@@ -80,22 +97,35 @@ function System(buses::Vector{Bus},
         push!(arrays, services)
     end
 
+    error_detected = false
+
     for component in Iterators.flatten(arrays)
-        add_component!(sys, component)
+        if add_component!(sys, component)
+            error_detected = true
+        end
```

Comment:
I don't understand what the logic is here? if the `component` is added successfully to `sys` (`add_component!(sys, component) == true`) then an error is detected? If this is actually how that function behaves then it requires renaming.

Reply:
I agree this was confusing, in latest commits add_component!() does not return a bool but instead throws an error and the above code is in a try/catch instead.